packages/central-server/src/apiV2/getCountryAccessList.js

```diff
@@ -18,29 +18,34 @@ const mapRequestsToEntities = (requestedEntities, projectCodeById) => {
 };
 
 export async function getCountryAccessList(req, res, next) {
-  const { userId, models } = req;
+  const { userId, models, query, accessPolicy } = req;
+  const { projectCode } = query;
+  if (!projectCode) throw new Error('No project code provided');
 
   try {
-    const countries = await models.entity.find({ type: 'country' });
-    const entityPermissions = await models.userEntityPermission.find({ user_id: userId });
-    const permittedEntityIds = new Set(entityPermissions.map(p => p.entity_id));
-    const accessRequests = await models.accessRequest.find({
-      user_id: userId,
-      processed_date: null,
-    });
-    const projects = await models.project.find();
-    const projectCodeById = reduceToDictionary(projects, 'id', 'code');
-    const entityRequests = mapRequestsToEntities(accessRequests, projectCodeById);
+    const project = await models.project.findOne({ code: projectCode });
+    if (!project) throw new Error(`No project found with code ‘${projectCode}’`);
+
+    const countries = await project.countries();
+    const countriesPendingAccess = (
+      await models.accessRequest.find({
+        user_id: userId,
+        project_id: project.id,
+        approved: null,
+      })
+    ).map(accessRequest => accessRequest.entity_id);
 
     const countryAccessList = countries
-      .filter(country => country.name !== 'No Country')
-      .map(country => ({
-        id: country.id,
-        name: country.name,
-        hasAccess: permittedEntityIds.has(country.id),
-        accessRequests: entityRequests[country.id] || [],
-        code: country.code,
-      }))
+      .reduce((listSoFar, country) => {
+        const { id, code, name } = country;
+        const hasAccess = project.permission_groups.some(permissionGroup =>
+          accessPolicy.allows(code, permissionGroup),
+        );
+        const hasPendingAccess = countriesPendingAccess.includes(id);
+
+        const countryAccessListItem = { id, name, hasAccess, hasPendingAccess };
```

Comment:
Slightly awkward variable name, but in the absence of TypeScript’s embrace, at least this will show up in results if a dev does a global search

(In the TS parts, the type for this response object is called `CountryAccessListItem`)